consensus/core/src/metrics.rs

```diff
@@ -800,3 +1179,813 @@ impl NodeMetrics {
         }
     }
 }
+
+pub(crate) struct ValidatorScoreMetrics {
+    pub(crate) stored: Vec<StoredScoreMetrics>,
+    pub(crate) cached: Vec<CachedScoreMetrics>,
+    pub(crate) score: Vec<Score>,
+}
+
+impl ValidatorScoreMetrics {
+    pub(crate) fn new(committee_size: usize) -> Self {
+        let stored = (0..committee_size)
+            .map(|_| StoredScoreMetrics::new())
+            .collect();
+        let cached = (0..committee_size)
+            .map(|_| CachedScoreMetrics::new())
+            .collect();
+        let score = (0..committee_size).map(|_| Score::new()).collect();
+        Self {
+            stored,
+            cached,
+            score,
+        }
+    }
+}
+
+pub(crate) struct Score(AtomicU64);
+impl Score {
+    pub(crate) fn new() -> Self {
+        Self(AtomicU64::new(0))
```

Comment:
I think this should be initialised with a perfect score assuming no prior misbehaviours
```suggestion
        Self(AtomicU64::new(u32::MAX))
```

Reply:
ah, of course!